server/src/index.js

```diff
@@ -27,30 +27,34 @@ const main = async () => {
       if (!await db.collection('Data').findOne({ url: data.url })) {
         const r = await db.collection('Data').insertOne(data)
         console.log('News added to the database.')
-        res.status(201).json({ success: true })
+        res.send(201)
       } else {
         console.log('Duplicate news found. Abort.')
-        res.status(409).json({ success: false })
+        res.send(409)
       }
     } else {
-      const parentId = ObjectID.createFromHexString(req.body.parent_id, (err, r) => {
-        if (err) {
+      try {
+        const parentId = ObjectID.createFromHexString(req.body.parentId)
+        if (await db.collection('Data').findOne({ _id: parentId })) {
+          const r = await db.collection('Data').insertOne(data)
+          await db.collection('Data').updateOne(
+            { _id: parentId },
+            { $push: { comment_ids: r.insertedId.toHexString() } })
+          console.log('Comment added to the database.')
+          res.send(201)
+        } else {
           console.log('Parent post not found. Abort.')
-          res.status(409).json({ success: false })
+          res.send(400)
+        }
+      } catch (err) {
+        if (req.body.parentId === null) {
```

Comment:
This check (whether the comment has a parent) should be outside. What if `ObjectID.createFromHexString(null)` is not an error?